sycl/include/syclcompat/math.hpp

```diff
@@ -307,8 +308,9 @@ inline T bfe(const T source, const uint32_t bit_start,
   static_assert(std::is_unsigned_v<T>);
   // FIXME(syclcompat-lib-reviewers): This ternary was added to catch a case
   // which may be undefined anyway. Consider that we are losing perf here.
-  const T mask =
-      num_bits >= CHAR_BIT * sizeof(T) ? T{-1} : ((T{1} << num_bits) - 1);
+  const T mask = num_bits >= CHAR_BIT * sizeof(T)
```

Comment:
I think you may prefer using `std::numeric_limits<unsigned char>::digits` here (from the `<limits>` C++ library) which should be equivalent to the `CHAR_BIT` definition from `<climits>`.

Reply:
Good point, cheers @GeorgeWeb!